typescript/lib/parsers/openai.ts

```diff
@@ -552,17 +553,22 @@ export class OpenAIChatModelParser extends ParameterizedModelParser<Chat.ChatCom
       const outputs: ExecuteResult[] = [];
       const responseWithoutChoices = _.omit(response, "choices");
       for (const choice of response.choices) {
+        const outputData: OutputDataWithValue | string | undefined =
+          buildOutputData(choice.message);
+        if (outputData === undefined) {
+          continue;
+        }
```

Comment:
```
if (!outputData) {
  continue;
}
```

to check for null or undefined? Unless this is intentional

Reply:
This is intentional. `outputData` can only be of type `OutputDataWithValue | string | undefined`. 

I did this because I did not want to set the prompt outputs to an empty array if there was no data (I feel it might be slightly unintuitive if you first serialize a prompt, don't run it, yet still see an empty `outputs` field. In our schema the `prompt.outputs` is optional so I think it makes more sense to simply not have present

Reply:
If outputData is null, we also don't want to serialize it, right? Or do we want to?

If we want the same behavior whether outputData is null or undefined, then we should change to `if (!outputData)`

Reply:
`outputData` can never be null since we set it to `OutputDataWithValue | string | undefined` but yea i'll add explicit check for null too!